barretenberg/cpp/src/barretenberg/relations/auxiliary_relation.hpp

```diff
@@ -155,55 +159,56 @@ template <typename FF_> class AuxiliaryRelationImpl {
          *            \_                                                                               _/
          *
          **/
-        auto limb_subproduct = w_1 * w_2_shift + w_1_shift * w_2;
-        auto non_native_field_gate_2 = (w_1 * w_4 + w_2 * w_3 - w_3_shift);
-        non_native_field_gate_2 *= LIMB_SIZE;
-        non_native_field_gate_2 -= w_4_shift;
-        non_native_field_gate_2 += limb_subproduct;
-        non_native_field_gate_2 *= q_4;
+        auto limb_subproduct = w_1_m * w_2_shift_m + w_1_shift_m * w_2_m;
+        auto non_native_field_gate_2_m = (w_1_m * w_4_m + w_2_m * w_3_m - w_3_shift_m);
+        non_native_field_gate_2_m *= LIMB_SIZE;
+        non_native_field_gate_2_m -= w_4_shift_m;
+        non_native_field_gate_2_m += limb_subproduct;
+        auto non_native_field_gate_2 = ShortAccumulator(non_native_field_gate_2_m) * ShortAccumulator(q_4_m);
 
         limb_subproduct *= LIMB_SIZE;
-        limb_subproduct += (w_1_shift * w_2_shift);
-        auto non_native_field_gate_1 = limb_subproduct;
-        non_native_field_gate_1 -= (w_3 + w_4);
-        non_native_field_gate_1 *= q_3;
+        limb_subproduct += (w_1_shift_m * w_2_shift_m);
+        auto non_native_field_gate_1_m = limb_subproduct;
+        non_native_field_gate_1_m -= (w_3_m + w_4_m);
+        auto non_native_field_gate_1 = ShortAccumulator(non_native_field_gate_1_m) * ShortAccumulator(q_3_m);
```

Comment:
will you add a brief comment as to why the `ShortAccumulator` transformation is required here?

Reply:
added comment